testing/test_decimal_approx.py

```diff
@@ -0,0 +1,16 @@
+from __future__ import annotations
+
+import decimal
+
+import pytest
+
+
+def test_decimal_approx_repr_issue():
+    trap = decimal.getcontext().traps[decimal.FloatOperation]
```

Comment:
use momkeypatch.setitem to ensure undo happens even on failures (not that i  expect a failure)